gensim/models/_utils_any2vec.pyx

```diff
@@ -7,11 +7,50 @@
 
 """General functions used for any2vec models."""
 
+from six import PY2
+import numpy as np
+cimport numpy as np
+
+
+cdef _byte_to_int_py3(b):
+    return b
+
+cdef _byte_to_int_py2(b):
+    return ord(b)
+
+_byte_to_int = _byte_to_int_py2 if PY2 else _byte_to_int_py3
+
+
 cpdef ft_hash(unicode string):
     """Calculate hash based on `string`.
     Reproduce `hash method from Facebook fastText implementation
     <https://github.com/facebookresearch/fastText/blob/master/src/dictionary.cc>`_.
 
+    Parameters
+    ----------
+    string : unicode
+        The string whose hash needs to be calculated.
+
+    Returns
+    -------
+    unsigned int
+        The hash of the string.
+
+    """
+    cdef unsigned int h = 2166136261
+    for c in string.encode("utf-8"):
```

Comment:
I think the encoding used by the hash is an implementation detail, so it shouldn't be visible to the user.

On the other hand, if they pass us text that cannot be encoded into utf-8 (for whatever reason, I'm finding it hard to imagine how), they will get an exception, which isn't good.

After discussion with @menshikh-iv, our options are:

- Ignore unicode errors in the hash
- Raise a helpful exception instead of the default UnicodeEncodeError
- Use the broken byte-based hash function as a fallback
- Do nothing, keep the code as-is
- Any others?

@piskvorky What do you think?